pandas/tests/indexes/test_setops.py

```diff
@@ -466,3 +466,19 @@ def test_setop_with_categorical(index, sort, method):
     result = getattr(index, method)(other[:5], sort=sort)
     expected = getattr(index, method)(index[:5], sort=sort)
     tm.assert_index_equal(result, expected)
+
+
+def test_intersection_duplicates_all_indexes(index):
+    # GH#38743
+    if index.empty:
+        # No duplicates in empty indexes
+        return
+
+    def check_intersection_commutative(left, right):
```

Comment:
ive been thinking we should make something like this of the form

```
@pytest.mark.parametrize("method", ["union", "intersection", "symmetric_difference"])
def test_setops_commute(method, index, index_fixture2):
    [...]
```

we've consolidated a lot of the behavior recently, but we still have some length-0 checks and self.equals(other) checks that seem a little wobbly

Reply:
Yeah, that sounds reasonable